platform-sdk/swirlds-common/src/main/java/com/swirlds/common/wiring/schedulers/SequentialThreadTask.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.common.wiring.schedulers;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.util.function.Consumer;
+
+/**
+ * A task that needs to be performed.
```

Comment:
Class doc isn't very specific, should perhaps mention sequential thread

Reply:
Originally this was an inline class, and so it was more obvious what it did. I've added the following.

```
A task that is performed by a {@link SequentialThreadTaskScheduler}.
```